06.bowling_object/game.rb

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require_relative 'frame'
+
+class Game
+  def self.calc_score(shot_scores)
+    frames = []
+    shot_scores.gsub('X', '10').split(',').each do |shot_score|
+      frames << Frame.new(frames.size) if frames.empty? || frames.last.finished?
+      frames.each do |frame|
+        frame.record_shot(shot_score.to_i)
```

Comment:
```suggestion
    shot_scores.split(',').each do |shot_score|
      frames << Frame.new(frames.size) if frames.empty? || frames.last.finished?
      frames.each do |frame|
        frame.record_shot(shot_score)
```

「Xは10である」というルールを知っておくべきは、Gameクラスなのか、Frameクラスなのか？
絶対にこっち、と断言できるわけではないですが、なんとなくボウリングに関する細かいルールはFrameクラスに寄せてしまって、Gameクラスはなるべくシンプルにした方が棲み分けが明確になりそうな気がします。

Reply:
Frameクラスに任せられそうなスコアの変換処理はFrameクラスで行うように変更しました